core_impl/src/test/java/com/google/instrumentation/stats/StatsTestUtil.java

```diff
@@ -0,0 +1,139 @@
+import com.google.common.base.Predicates;
+import com.google.common.collect.Iterables;
+import com.google.common.truth.Truth;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/** Stats test utilities. */
+final class StatsTestUtil {
+
+  private StatsTestUtil() {}
+
+  static StatsContextImpl createContext(
+      StatsContextFactoryImpl factory, TagKey key, TagValue value) {
+    return createContext(factory, Tag.create(key, value));
+  }
+
+  static StatsContextImpl createContext(
+      StatsContextFactoryImpl factory, TagKey key1, TagValue value1, TagKey key2, TagValue value2) {
+    return createContext(factory, Tag.create(key1, value1), Tag.create(key2, value2));
+  }
+
+  /**
+   * Creates a {@code StatsContextImpl} from a factory and a list tags.
+   *
+   * @param factory the factory used to produce the {@code StatsContextImpl}.
+   * @param tags a list of tags to add to the {@code StatsContextImpl}.
+   * @return a {@code StatsContextImpl} with the given tags.
+   */
+  private static StatsContextImpl createContext(
+      StatsContextFactoryImpl factory, Tag... tags) {
+    StatsContextImpl.Builder builder = factory.getDefault().builder();
+    for (Tag tag : tags) {
+      builder.set(tag.getKey(), tag.getValue());
+    }
+    return builder.build();
+  }
+
+  /**
+   * Creates a {@code DistributionAggregation} by adding the given values to a new {@link
+   * MutableDistribution}.
+   *
+   * @param tags the {@code DistributionAggregation}'s tags.
+   * @param bucketBoundaries the bucket boundaries.
+   * @param values the values to add to the distribution.
+   * @return the new {@code DistributionAggregation}
+   */
+  static DistributionAggregation createDistributionAggregation(
+      List<Tag> tags, BucketBoundaries bucketBoundaries, List<Double> values) {
+    MutableDistribution mdist = MutableDistribution.create(bucketBoundaries);
+    for (double value : values) {
+      mdist.add(value);
+    }
+    MutableDistribution.Range range = mdist.getRange();
+    return DistributionAggregation.create(
+        mdist.getCount(),
+        mdist.getMean(),
+        mdist.getSum(),
+        DistributionAggregation.Range.create(range.getMin(), range.getMax()),
+        tags,
+        mdist.getBucketCounts());
+  }
+
+  /**
+   * Asserts that the two sets of {@code DistributionAggregation}s are equivalent, with a given
+   * tolerance. The tolerance is used when comparing the mean and sum of values. The order of the
+   * {@code DistributionAggregation}s has no effect. The expected parameter is last, because it is
+   * likely to be a larger expression.
+   *
+   * @param tolerance the tolerance used for {@code double} comparison.
+   * @param actual the actual test result.
+   * @param expected the expected value.
+   * @throws AssertionError if the {@code DistributionAggregation}s don't match.
+   */
+  static void assertDistributionAggregationsEquivalent(
+      double tolerance,
```

Comment:
Minor: I think we could define and use `TOLERANCE` in `StatsTestUtil` so that `assertDistributionAggregationsEquivalent()` has fewer arguments and is easier to use. I don't think it's quite necessary to "customize" the tolerance when calling `assertDistributionAggregationsEquivalent()`, and even it's necessary, we can overload this method (one with the parameter `tolerance` and one without).